File: PythonCode/lintcode/easy/KthPrimeNumber.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

# @Time    : 2018/2/8 9:48
# @Author  : bearyang
# @Site    : 
# @File    : KthPrimeNumber.py
# @Software: PyCharm

import math
class Solution:
    def kthPrime(self, n):
        index = 0
        for i in range(2, n+1):
            if self.isPrime(i):
                index += 1
        return index


    def isPrime(self, n):
        """
        判断一个数是否为素数
        :param n:
        :return:
        """
        if n == 0 or n == 1:
            return False
        for i in range(2, int(math.sqrt(n)) + 1):
            if n % i == 0:
                return False
        return True


test = Solution()
print(test.kthPrime(2))
print(test.kthPrime(3))
print(test.kthPrime(5))
print(test.kthPrime(7))
print(test.kthPrime(11))

File: JsCode/lintcode/easy/KthPrimeNumber.js
/**
 * 792. Kth Prime Number
 * http://www.lintcode.com/zh-cn/problem/kth-prime-number/
 * 判断素数n是第几个素数
 */
var kthPrimeNumber = function (n) {
    var index = 0;
    for (var i = 2; i <= n; i++) {
        if (isPrime(i)) {
            index++;
        }
    }
    return index;
};

/**
 * 判断一个数是否为素数
 * @param n
 * @returns {boolean}
 */
var isPrime = function (n) {
    if ( n === 0 || n === 1){
        return false;
    }
    for (var i = 2; i <= Math.sqrt(n); i++) {
        if (n % i === 0) {
            return false;
        }
    }
    return true;
};

console.log(kthPrimeNumber(2));
console.log(kthPrimeNumber(3));
console.log(kthPrimeNumber(5));
console.log(kthPrimeNumber(7));
console.log(kthPrimeNumber(19));